src/unix/syscall.c

```diff
@@ -283,6 +283,45 @@ static inline tuple resolve_cstring(tuple cwd, const char *f)
     return t;
 }
 
+static inline tuple resolve_utf8(tuple cwd, const char *f)
+{
+    tuple t = *f == '/' ? filesystem_getroot(current->p->fs) : cwd;
+
+    buffer a = little_stack_buffer(3 * NAME_MAX);
+    char y;
+    int nbytes;
+
+    if (buffer_length(a)) {
+        t = lookup(t, intern(a));
+    }
```

Comment:
Here, the buffer has just been allocated and is guaranteed to be empty (buffer_length() will always return 0), so I would remove this if() block.

Reply:
This one is _definitely_ a copy-paste error, it's the same block as at the end of the function. I have no good excuse for why it made it into a PR :(.